### PR TITLE
Kinetics dataloader & model changes for HW implementation

### PR DESCRIPTION
**Pull request by alicangok**
For image pairs, the updated dataloader now folds each image first, then concatenates the two.

This change was necessary to fill the data memory of the accelerator in a more efficient manner, as used in the demo.

~

The model now correctly uses ai8x.add modules instead of simple additions. (Using simple additions without clamping had resulted in ~1% loss in accuracy when evaluating with 8-bit-mode)

### CODE REVIEW

**Review by ermanok — APPROVED** [review]
Looks good.

**Review by rotx-maxim — CHANGES_REQUESTED** [review]
Should we change the for loops to
images_concat = [torch.cat((x, y, dim=0)) for x, y in zip(images, images[1:])]
and
images_concat = [np.concatenate((x, y, axis=2)) for x, y in zip(images, images[1:])]

**Comment by alicangok** [comment]
> Should we change the for loops to images_concat = [torch.cat((x, y, dim=0)) for x, y in zip(images, images[1:])] and images_concat = [np.concatenate((x, y, axis=2)) for x, y in zip(images, images[1:])]

Sure. `For` loops come more naturally to me than list comprehensions, but we could make that change for a marginal speed improvement.

**Review by rotx-maxim — APPROVED** [review]
